config/fcos/v1_3_exp/translate.go

```diff
@@ -37,3 +53,173 @@ func (c Config) ToIgn3_2(options common.TranslateOptions) (types.Config, report.
+		}
+
+		// create RAIDs
+		raidDevices := func(labelPrefix string) []types.Device {
+			count := len(rendered.Storage.Disks)
+			ret := make([]types.Device, count)
+			for i := 0; i < count; i++ {
+				ret[i] = types.Device(fmt.Sprintf("/dev/disk/by-partlabel/%s-%d", labelPrefix, i+1))
+			}
+			return ret
+		}
+		rendered.Storage.Raid = []types.Raid{{
+			Devices: raidDevices("esp"),
+			Level:   "raid1",
+			Name:    "md-esp",
+			// put the RAID superblock at the end of the
+			// partition to avoid confusing UEFI
+			Options: []types.RaidOption{"--metadata=1.0"},
+		}, {
+			Devices: raidDevices("boot"),
+			Level:   "raid1",
+			Name:    "md-boot",
+			// put the RAID superblock at the end of the
+			// partition so BIOS GRUB doesn't need to
+			// understand RAID
+			Options: []types.RaidOption{"--metadata=1.0"},
+		}, {
+			Devices: raidDevices("root"),
+			Level:   "raid1",
+			Name:    "md-root",
+		}}
+		renderedTranslations.AddFromCommonSource(path.New("yaml", "boot_device", "mirror"), path.New("json", "storage", "raid"), rendered.Storage.Raid)
+
+		// create filesystems, except for root
+		rendered.Storage.Filesystems = []types.Filesystem{{
+			Device:         "/dev/md/md-esp",
+			Format:         util.StrToPtr("vfat"),
+			Label:          util.StrToPtr("EFI-SYSTEM"),
+			WipeFilesystem: util.BoolToPtr(true),
+		}, {
+			Device:         "/dev/md/md-boot",
+			Format:         util.StrToPtr("ext4"),
+			Label:          util.StrToPtr("boot"),
+			WipeFilesystem: util.BoolToPtr(true),
+		}}
+		renderedTranslations.AddFromCommonSource(path.New("yaml", "boot_device", "mirror"), path.New("json", "storage", "filesystems"), rendered.Storage.Filesystems)
+	}
+
+	// encrypted root partition
+	if wantLuks {
+		luksDevice := "/dev/disk/by-partlabel/root"
+		if wantMirror {
+			luksDevice = "/dev/md/md-root"
+		}
+		clevis, ts2, r2 := translateBootDeviceLuks(c.BootDevice.Luks, options)
+		rendered.Storage.Luks = []types.Luks{{
+			Clevis:     &clevis,
+			Device:     &luksDevice,
+			Label:      util.StrToPtr("luks-root"),
+			Name:       "root",
+			WipeVolume: util.BoolToPtr(true),
+		}}
+		lpath := path.New("yaml", "boot_device", "luks")
+		rpath := path.New("json", "storage", "luks", 0)
+		renderedTranslations.Merge(ts2.PrefixPaths(lpath, rpath.Append("clevis")))
+		for _, f := range []string{"device", "label", "name", "wipeVolume"} {
+			renderedTranslations.AddTranslation(lpath, rpath.Append(f))
+		}
+		r.Merge(r2)
+	}
+
+	// create root filesystem
+	var rootDevice string
+	switch {
+	case wantLuks:
+		// LUKS, or LUKS on RAID
+		rootDevice = "/dev/disk/by-id/dm-name-root"
+	case wantMirror:
+		// RAID without LUKS
+		rootDevice = "/dev/md/md-root"
+	default:
+		panic("can't happen")
+	}
+	rootFilesystem := types.Filesystem{
+		Device:         rootDevice,
+		Format:         util.StrToPtr("xfs"),
+		Label:          util.StrToPtr("root"),
+		WipeFilesystem: util.BoolToPtr(true),
+	}
```

Comment:
This just made me realize something. In cosa, [we enable reflinks](https://github.com/coreos/coreos-assembler/blob/7bf455a17d7033facbf6207c8d93557415f64aee/src/create_disk.sh#L239). In tests and documentation that reformat the rootfs to XFS, we're not re-adding it. Ideally, there'd be a way to just "copy" the settings from the rootfs we nuked, but short of that we should include it in our docs and make sure we test it too.

Anyway, circling back to here, since it's sugar, I think we should keep carry our opinion forward of enabling reflinks to match the original configuration.

Reply:
I agree that we should replicate any unusual FS options into the desugared config.  `mkfs.xfs(8)` on both Fedora 33 and CentOS 8 says that reflinks are enabled by default, so I don't think we need to do anything here?

Reply:
Ahh nice, didn't realize that was the default now. That means we should be able to drop it from cosa too. Will add that to the list.